File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    id: int
    user_id: int
    original_filename: str
    stored_filename: str
    content_type: str | None

    status: str
    error_message: str | None
    processed_at: datetime | None
    created_at: datetime

    chunk_count: int

    class Config:
        from_attributes = True


class DocumentResponse(BaseModel):
    id: int
    user_id: int
    original_filename: str
    stored_filename: str
    content_type: str | None

    chunk_count: int

    status: str
    error_message: str | None
    processed_at: datetime | None
    created_at: datetime

    class Config:
        from_attributes = True


class DocumentChunkResponse(BaseModel):
    id: int
    document_id: int
    chunk_index: int
    content: str
    created_at: datetime

    class Config:
        from_attributes = True


class DocumentSearchResponse(BaseModel):
    id: int
    document_id: int
    document_filename: str
    chunk_index: int
    content: str
    similarity: float
    created_at: datetime


class DocumentDeleteResponse(BaseModel):
    message: str
    document_id: int

class DocumentRetryResponse(BaseModel):
    message: str
    document_id: int
    status: str

File: frontend/src/types/document.ts
export type DocumentStatus =
  | "UPLOADED"
  | "PROCESSING"
  | "COMPLETED"
  | "FAILED";


export interface Document {
  id: number;
  user_id: number;
  original_filename: string;
  stored_filename: string;
  content_type: string | null;

  chunk_count: number | null;

  status: DocumentStatus;
  error_message: string | null;
  processed_at: string | null;
  created_at: string;
}

// 새 업로드 API 응답은 문서(Document) 자체이다.
export type DocumentUploadResponse = Document;


export interface DocumentChunk {
  id: number;
  document_id: number;
  chunk_index: number;
  content: string;
  created_at: string;
}


export interface DocumentDeleteResponse {
  message: string;
  document_id: number;
}


export interface DocumentSearchResult {
  id: number;
  document_id: number;
  document_filename: string;
  chunk_index: number;
  content: string;
  similarity: number;
  created_at: string;
}

export interface DocumentRetryResponse {
  message: string;
  document_id: number;
  status: DocumentStatus;
}